.github/workflows/release_apps.yml

```diff
@@ -216,25 +203,218 @@ jobs:
+      - name: Generate oxfmt JS build
+        working-directory: apps/oxfmt
+        run: pnpm run build-js
+
+      - name: Generate npm packages
+        run: |
+          node npm/oxfmt/scripts/generate-packages.js
+          cat npm/oxfmt/package.json
+          for package in npm/oxfmt*; do cat $package/package.json ; echo ; done
+
+      - run: npm install -g npm@latest # For trusted publishing support
+
+      - name: Publish npm packages as latest
+        # NOTE: The trailing slash on $package/ changes it to publishing the directory
+        run: |
+          for package in npm/oxfmt*
+          do
+            pnpm publish $package/ --provenance --access public --no-git-checks
+            echo '----'
+          done
+
+  github-release:
+    name: GitHub Release
+    needs: [check, build-oxlint, build-oxfmt, publish-oxlint, publish-oxfmt]
+    permissions:
+      contents: write # for softprops/action-gh-release@v1.3.1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          token: ${{ secrets.OXC_BOT_PAT }}
+          fetch-depth: 0 # for changelog
+          persist-credentials: true
+
+      - uses: oxc-project/setup-rust@f03aedf4620c1d3aa169a2510e24787d362bbe0b # v1.0.8
+        with:
+          restore-cache: false
+          tools: cargo-release-oxc
+
+      - name: Run
+        id: run
+        run: |
+          mkdir -p target
+          cargo release-oxc changelog --release oxfmt --release oxlint
+          {
+            echo '# Oxlint'
+            echo
+            cat ./target/OXLINT_CHANGELOG
+            echo '# Oxfmt'
+            echo
+            cat ./target/OXFMT_CHANGELOG
+          } > ./target/RELEASE_BODY.md
+
+      - name: tag oxlint and oxfmt
+        env:
+          NEEDS_CHECK_OUTPUTS_OXFMT_VERSION: ${{ needs.check.outputs.oxfmt_version }}
+          NEEDS_CHECK_OUTPUTS_OXLINT_VERSION: ${{ needs.check.outputs.oxlint_version }}
+        run: |
+          git tag oxfmt_v${NEEDS_CHECK_OUTPUTS_OXFMT_VERSION} ${{ github.sha }}
+          git tag oxlint_v${NEEDS_CHECK_OUTPUTS_OXLINT_VERSION} ${{ github.sha }}
+          git push origin oxlint_v${NEEDS_CHECK_OUTPUTS_OXLINT_VERSION} oxfmt_v${NEEDS_CHECK_OUTPUTS_OXFMT_VERSION}
+
       - name: Create GitHub Release
         uses: softprops/action-gh-release@6da8fa9354ddfdc4aeace5fc48d7f679b5214090 # v2.4.1
         with:
-          body: ${{ steps.run.outputs.CHANGELOG }}
+          body_path: ./target/RELEASE_BODY.md
           draft: false
-          name: oxlint v${{ needs.check.outputs.version }}
-          tag_name: oxlint_v${{ needs.check.outputs.version }}
+          name: oxlint v${{ needs.check.outputs.oxlint_version }} & oxfmt v${{ needs.check.outputs.oxfmt_version }}
+          tag_name: apps_v${{ needs.check.outputs.oxlint_version }}
           target_commitish: ${{ github.sha }}
 
       - name: wait 3 minutes for smoke test
         run: sleep 180s
 
   smoke:
```

Comment:
This will fail because:

- `bash` does not exists on Alpine
- `oxlint@${OXLINT_VERSION}` is invalid on Windows

refs: #15360